File: Go/XServer/config_test.go
// Copyright (c) 2026 FrameworkEase Technologies. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

package XServer

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestConfigure(t *testing.T) {
	Configure(50051, 50052)
	assert.Equal(t, 50051, configuredServePort, "ServePort 配置值应当为 50051。")
	assert.Equal(t, 50052, configuredProbePort, "ProbePort 配置值应当为 50052。")
}

File: CSharp/tests~/Assets/Tests/Runtime/Scripts/XServer/config_test.cs
// Copyright (c) 2026 FrameworkEase Technologies. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

#if !UNITY_5_3_OR_NEWER
using FrameworkEase.Universal.X.gRPC;
using NUnit.Framework;

public partial class TestXServer
{
    [Test]
    public void Configure()
    {
        XServer.Configure(50051, 50052);
        Assert.That(XServer.Configuration.ServePort, Is.EqualTo(50051), "ServePort 配置值应当为 50051。");
        Assert.That(XServer.Configuration.ProbePort, Is.EqualTo(50052), "ProbePort 配置值应当为 50052。");
    }
}
#endif
